modules/ocf_printhost/files/enforcer

```diff
@@ -0,0 +1,182 @@
+is to add and subtract remaining pages as user jobs are processed.
+
+When a user sends a job to the print server, Tea4CUPS passes the job as a
+PostScript file to enforcer before sending it to the printer. Enforcer calls
+Ghostscript to count pages and connects to mysql to check against the user's
+quota. If enforcer returns 255 (a.k.a. -1), the job is rejected. Otherwise,
+enforcer gets callued again when the job is done and logs the job in mysql,
+taking care to set the page count to zero if an error was encountered.
+
+Another function of enforcer is to send notifications to desktops to let users
+know when a job has been sent to the printer, been rejected due to the quota,
+finished printing, or failed to print.
+"""
+import os
+from argparse import ArgumentParser
+from collections import namedtuple
+from configparser import ConfigParser
+from datetime import datetime
+from subprocess import PIPE
+from subprocess import Popen
+from sys import argv
+from sys import exit
+from syslog import syslog
+from textwrap import dedent
+from traceback import format_exc
+
+import ocflib.printing.quota as quota
+from ocflib.constants import MAIL_SIGNATURE
+from ocflib.misc.mail import send_mail_user
+from ocflib.misc.mail import send_problem_report
+
+
+CONF_FILE = '/opt/share/enforcer/enforcer.conf'
+
+
+Message = namedtuple('Message', ['subject', 'body'])
+
+
+INSUFFICIENT_QUOTA_MESSAGE = Message(
+    subject='[OCF] Your latest print job was rejected',
+    body=dedent("""\
+        Greetings from the Open Computing Facility,
+
+        This email is letting you know that your most recent print job was
+        rejected since it would exceed your daily quota. The daily quota is
+        {daily_quota} pages today and the semesterly quota is {semester_quota} pages.
+
+        Username: {user}
+        Time: {time}
+        File: {doc_name}
+        Total pages: {pages}
+        Pages left today: {daily_pages}
+        Pages left this semester: {semester_pages}
+
+
+        Does something look wrong? Please reply to
+
+            help@ocf.berkeley.edu
+
+
+        """) + MAIL_SIGNATURE
+)
+
+PRINTER_ERROR_MESSAGE = Message(
+    subject='[OCF] Your latest print job failed',
+    body=dedent("""\
+        Greetings from the Open Computing Facility,
+
+        This email is from the OCF to let you know that your most recent print
+        job failed due to a printer error. If there's something wrong with the
+        printers, please alert the publications staff at the desk.
+
+        Username: {user}
+        Time: {time}
+        File: {doc_name}
+        Total pages: {pages}
+        Pages left today: {daily_pages}
+        Pages left this semester: {semester_pages}
+
+
+        Still can't get it to print? Please reply to
+
+            help@ocf.berkeley.edu
+
+
+        """) + MAIL_SIGNATURE
+)
+
+
+def read_config():
+    conf = ConfigParser()
+    conf.readfp(open(CONF_FILE))
+    mysql_host = conf.get('enforcer', 'user')
```

Comment:
`mysql_host` comes from `enforcer->user`?

(noticed this last night too, the pollution file is mislabeled as well)